iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/CreateCouplingDevice.java

```diff
@@ -109,51 +109,73 @@ public void apply(Network network, boolean throwException, ComputationManager co
-                    // otherwise the coupler is on each side attached to all busbars of the corresponding section
-                    int nbOpenDisconnectors = 0;
-                    if (bbsList1.size() != 1 || position1.getSectionIndex() != position2.getSectionIndex()) {
-                        nbOpenDisconnectors = bbsList1.size() * 2;
-                        createTopologyFromBusbarSectionList(voltageLevel1, breakerNode1, switchPrefixId, bbsList1);
-                        createTopologyFromBusbarSectionList(voltageLevel2, breakerNode2, switchPrefixId, bbsList2);
-                        LOGGER.info("{} open disconnectors created on parallel busbar section in voltage level {}", nbOpenDisconnectors, voltageLevel1.getId());
-                        openDisconnectorsAddedReport(reporter, voltageLevel1.getId(), nbOpenDisconnectors);
-                    }
-                } else {
-                    LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs2.getId());
-                    noBusbarSectionPositionExtensionReport(reporter, bbs2);
-                }
-            } else {
-                LOGGER.warn("No busbar section position extension found on {}, only one disconnector is created.", bbs1.getId());
-                noBusbarSectionPositionExtensionReport(reporter, bbs1);
-            }
+            applyOnBusbarSections(voltageLevel1, voltageLevel2, bbs1, bbs2, reporter);
         }
         LOGGER.info("New coupling device was added to voltage level {} between {} and {}", voltageLevel1.getId(), busOrBbs1, busOrBbs2);
         newCouplingDeviceAddedReport(reporter, voltageLevel1.getId(), busOrBbsId1, busOrBbsId2);
     }
 
+    /**
+     * Apply the modification on the two specified busbar sections
+     */
+    private void applyOnBusbarSections(VoltageLevel voltageLevel1, VoltageLevel voltageLevel2, BusbarSection bbs1, BusbarSection bbs2, Reporter reporter) {
+        // busbar sections exist: voltage level is NODE_BREAKER
+        int breakerNode1 = voltageLevel1.getNodeBreakerView().getMaximumNodeIndex() + 1;
+        int breakerNode2 = breakerNode1 + 1;
+        int nbOpenDisconnectors = 0;
+
+        // Breaker
+        createNBBreaker(breakerNode1, breakerNode2, "", switchPrefixId, voltageLevel1.getNodeBreakerView(), false);
+
+        // Disconnectors
+        BusbarSectionPosition position1 = bbs1.getExtension(BusbarSectionPosition.class);
+        BusbarSectionPosition position2 = bbs2.getExtension(BusbarSectionPosition.class);
+        boolean checkAndFilterOtherBbs = position1 != null && position2 != null && position1.getSectionIndex() == position2.getSectionIndex();
+        if (position1 != null) {
+            // List of the bars for the first section
+            List<BusbarSection> bbsList1 = voltageLevel1.getNodeBreakerView().getBusbarSectionStream()
```

Comment:
Here it's probably worth it to extract a method to reuse for the other side

Reply:
changed